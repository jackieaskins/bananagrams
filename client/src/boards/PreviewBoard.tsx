import { Box, Card } from "@chakra-ui/react";
import { convertToArray } from "./convert";
import { Board } from "./types";

type PreviewBoardProps = {
  board: Board;
  tileSize: number;
};

export default function PreviewBoard({
  board,
  tileSize,
}: PreviewBoardProps): JSX.Element {
  const tilePixels = `${tileSize}px`;

  return (
    <Card display="inline-flex" flexDirection="column" variant="outline">
      {convertToArray(board).map((row, x) => (
        <Box key={x} display="flex">
          {row.map((boardSquare, y) => {
            const tile = boardSquare?.tile ?? null;

            if (tile === null) {
              return <Box key={y} height={tilePixels} width={tilePixels} />;
            }

            return (
              <img
                key={y}
                src={`/assets/images/${tile.letter}.png`}
                alt={`Tile for the letter ${tile.letter}`}
                width={tilePixels}
                height={tilePixels}
              />
            );
          })}
        </Box>
      ))}
    </Card>
  );
}
