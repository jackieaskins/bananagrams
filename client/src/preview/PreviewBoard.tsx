import { Board, getSquareId } from '../boards/types';
import { getBoardBoundaries } from '../boards/validate';
import TransparentCard from '../card/TransparentCard';
import PreviewTile from './PreviewTile';
import { TILE_SIZE } from './constants';

type PreviewBoardProps = {
  board: Board;
};

const PreviewBoard = ({ board }: PreviewBoardProps): JSX.Element => {
  const { rowStart, rowEnd, colStart, colEnd } = getBoardBoundaries(board);

  const boardRows = [];

  if (
    rowStart != null &&
    rowEnd != null &&
    colStart != null &&
    colEnd != null
  ) {
    for (let row = rowStart; row <= rowEnd; row++) {
      const boardRow = [];
      for (let col = colStart; col <= colEnd; col++) {
        const id = getSquareId({ row, col });
        const square = board[id];

        if (square) {
          boardRow.push(
            <PreviewTile
              key={square.tile.id}
              id={square.tile.id}
              letter={square.tile.letter}
            />
          );
        } else {
          boardRow.push(
            <div
              key={id}
              css={{ height: `${TILE_SIZE}px`, width: `${TILE_SIZE}px` }}
            />
          );
        }
      }

      boardRows.push(
        <div key={row} css={{ display: 'flex' }}>
          {boardRow}
        </div>
      );
    }
  }

  return (
    <TransparentCard
      bodyStyle={{ padding: 0 }}
      cardCSS={{
        width: '100%',
        display: 'flex',
        alignItems: 'center',
        justifyContent: 'center',
        flexGrow: 1,
        margin: '5px 0',
      }}
    >
      <div>{boardRows}</div>
    </TransparentCard>
  );
};

export default PreviewBoard;
