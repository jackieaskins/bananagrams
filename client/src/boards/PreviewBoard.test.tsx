import { shallow } from 'enzyme';

import { boardSquareFixture } from '../fixtures/board';
import PreviewBoard from './PreviewBoard';

describe('<PreviewBoard />', () => {
  it('renders properly', () => {
    expect(
      shallow(
        <PreviewBoard
          board={[
            [boardSquareFixture(), null],
            [null, null],
          ]}
          tileSize={23}
        />
      )
    ).toMatchSnapshot();
  });
});
