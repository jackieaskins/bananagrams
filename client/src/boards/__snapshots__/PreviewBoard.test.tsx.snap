// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PreviewBoard /> renders properly 1`] = `
<ForwardRef(Card2)
  display="inline-flex"
  flexDirection="column"
  variant="outline"
>
  <Box
    display="flex"
    key="0"
  >
    <img
      alt="Tile for the letter A"
      height="23px"
      key="0"
      src="/assets/images/A.png"
      width="23px"
    />
    <Box
      height="23px"
      key="1"
      width="23px"
    />
    <Box
      height="23px"
      key="2"
      width="23px"
    />
    <Box
      height="23px"
      key="3"
      width="23px"
    />
    <Box
      height="23px"
      key="4"
      width="23px"
    />
    <Box
      height="23px"
      key="5"
      width="23px"
    />
    <Box
      height="23px"
      key="6"
      width="23px"
    />
    <Box
      height="23px"
      key="7"
      width="23px"
    />
    <Box
      height="23px"
      key="8"
      width="23px"
    />
    <Box
      height="23px"
      key="9"
      width="23px"
    />
    <Box
      height="23px"
      key="10"
      width="23px"
    />
    <Box
      height="23px"
      key="11"
      width="23px"
    />
    <Box
      height="23px"
      key="12"
      width="23px"
    />
    <Box
      height="23px"
      key="13"
      width="23px"
    />
    <Box
      height="23px"
      key="14"
      width="23px"
    />
    <Box
      height="23px"
      key="15"
      width="23px"
    />
    <Box
      height="23px"
      key="16"
      width="23px"
    />
    <Box
      height="23px"
      key="17"
      width="23px"
    />
    <Box
      height="23px"
      key="18"
      width="23px"
    />
    <Box
      height="23px"
      key="19"
      width="23px"
    />
    <Box
      height="23px"
      key="20"
      width="23px"
    />
  </Box>
  <Box
    display="flex"
    key="1"
  >
    <Box
      height="23px"
      key="0"
      width="23px"
    />
    <Box
      height="23px"
      key="1"
      width="23px"
    />
    <Box
      height="23px"
      key="2"
      width="23px"
    />
    <Box
      height="23px"
      key="3"
      width="23px"
    />
    <Box
      height="23px"
      key="4"
      width="23px"
    />
    <Box
      height="23px"
      key="5"
      width="23px"
    />
    <Box
      height="23px"
      key="6"
      width="23px"
    />
    <Box
      height="23px"
      key="7"
      width="23px"
    />
    <Box
      height="23px"
      key="8"
      width="23px"
    />
    <Box
      height="23px"
      key="9"
      width="23px"
    />
    <Box
      height="23px"
      key="10"
      width="23px"
    />
    <Box
      height="23px"
      key="11"
      width="23px"
    />
    <Box
      height="23px"
      key="12"
      width="23px"
    />
    <Box
      height="23px"
      key="13"
      width="23px"
    />
    <Box
      height="23px"
      key="14"
      width="23px"
    />
    <Box
      height="23px"
      key="15"
      width="23px"
    />
    <Box
      height="23px"
      key="16"
      width="23px"
    />
    <Box
      height="23px"
      key="17"
      width="23px"
    />
    <Box
      height="23px"
      key="18"
      width="23px"
    />
    <Box
      height="23px"
      key="19"
      width="23px"
    />
    <Box
      height="23px"
      key="20"
      width="23px"
    />
  </Box>
  <Box
    display="flex"
    key="2"
  >
    <Box
      height="23px"
      key="0"
      width="23px"
    />
    <Box
      height="23px"
      key="1"
      width="23px"
    />
    <Box
      height="23px"
      key="2"
      width="23px"
    />
    <Box
      height="23px"
      key="3"
      width="23px"
    />
    <Box
      height="23px"
      key="4"
      width="23px"
    />
    <Box
      height="23px"
      key="5"
      width="23px"
    />
    <Box
      height="23px"
      key="6"
      width="23px"
    />
    <Box
      height="23px"
      key="7"
      width="23px"
    />
    <Box
      height="23px"
      key="8"
      width="23px"
    />
    <Box
      height="23px"
      key="9"
      width="23px"
    />
    <Box
      height="23px"
      key="10"
      width="23px"
    />
    <Box
      height="23px"
      key="11"
      width="23px"
    />
    <Box
      height="23px"
      key="12"
      width="23px"
    />
    <Box
      height="23px"
      key="13"
      width="23px"
    />
    <Box
      height="23px"
      key="14"
      width="23px"
    />
    <Box
      height="23px"
      key="15"
      width="23px"
    />
    <Box
      height="23px"
      key="16"
      width="23px"
    />
    <Box
      height="23px"
      key="17"
      width="23px"
    />
    <Box
      height="23px"
      key="18"
      width="23px"
    />
    <Box
      height="23px"
      key="19"
      width="23px"
    />
    <Box
      height="23px"
      key="20"
      width="23px"
    />
  </Box>
  <Box
    display="flex"
    key="3"
  >
    <Box
      height="23px"
      key="0"
      width="23px"
    />
    <Box
      height="23px"
      key="1"
      width="23px"
    />
    <Box
      height="23px"
      key="2"
      width="23px"
    />
    <Box
      height="23px"
      key="3"
      width="23px"
    />
    <Box
      height="23px"
      key="4"
      width="23px"
    />
    <Box
      height="23px"
      key="5"
      width="23px"
    />
    <Box
      height="23px"
      key="6"
      width="23px"
    />
    <Box
      height="23px"
      key="7"
      width="23px"
    />
    <Box
      height="23px"
      key="8"
      width="23px"
    />
    <Box
      height="23px"
      key="9"
      width="23px"
    />
    <Box
      height="23px"
      key="10"
      width="23px"
    />
    <Box
      height="23px"
      key="11"
      width="23px"
    />
    <Box
      height="23px"
      key="12"
      width="23px"
    />
    <Box
      height="23px"
      key="13"
      width="23px"
    />
    <Box
      height="23px"
      key="14"
      width="23px"
    />
    <Box
      height="23px"
      key="15"
      width="23px"
    />
    <Box
      height="23px"
      key="16"
      width="23px"
    />
    <Box
      height="23px"
      key="17"
      width="23px"
    />
    <Box
      height="23px"
      key="18"
      width="23px"
    />
    <Box
      height="23px"
      key="19"
      width="23px"
    />
    <Box
      height="23px"
      key="20"
      width="23px"
    />
  </Box>
  <Box
    display="flex"
    key="4"
  >
    <Box
      height="23px"
      key="0"
      width="23px"
    />
    <Box
      height="23px"
      key="1"
      width="23px"
    />
    <Box
      height="23px"
      key="2"
      width="23px"
    />
    <Box
      height="23px"
      key="3"
      width="23px"
    />
    <Box
      height="23px"
      key="4"
      width="23px"
    />
    <Box
      height="23px"
      key="5"
      width="23px"
    />
    <Box
      height="23px"
      key="6"
      width="23px"
    />
    <Box
      height="23px"
      key="7"
      width="23px"
    />
    <Box
      height="23px"
      key="8"
      width="23px"
    />
    <Box
      height="23px"
      key="9"
      width="23px"
    />
    <Box
      height="23px"
      key="10"
      width="23px"
    />
    <Box
      height="23px"
      key="11"
      width="23px"
    />
    <Box
      height="23px"
      key="12"
      width="23px"
    />
    <Box
      height="23px"
      key="13"
      width="23px"
    />
    <Box
      height="23px"
      key="14"
      width="23px"
    />
    <Box
      height="23px"
      key="15"
      width="23px"
    />
    <Box
      height="23px"
      key="16"
      width="23px"
    />
    <Box
      height="23px"
      key="17"
      width="23px"
    />
    <Box
      height="23px"
      key="18"
      width="23px"
    />
    <Box
      height="23px"
      key="19"
      width="23px"
    />
    <Box
      height="23px"
      key="20"
      width="23px"
    />
  </Box>
  <Box
    display="flex"
    key="5"
  >
    <Box
      height="23px"
      key="0"
      width="23px"
    />
    <Box
      height="23px"
      key="1"
      width="23px"
    />
    <Box
      height="23px"
      key="2"
      width="23px"
    />
    <Box
      height="23px"
      key="3"
      width="23px"
    />
    <Box
      height="23px"
      key="4"
      width="23px"
    />
    <Box
      height="23px"
      key="5"
      width="23px"
    />
    <Box
      height="23px"
      key="6"
      width="23px"
    />
    <Box
      height="23px"
      key="7"
      width="23px"
    />
    <Box
      height="23px"
      key="8"
      width="23px"
    />
    <Box
      height="23px"
      key="9"
      width="23px"
    />
    <Box
      height="23px"
      key="10"
      width="23px"
    />
    <Box
      height="23px"
      key="11"
      width="23px"
    />
    <Box
      height="23px"
      key="12"
      width="23px"
    />
    <Box
      height="23px"
      key="13"
      width="23px"
    />
    <Box
      height="23px"
      key="14"
      width="23px"
    />
    <Box
      height="23px"
      key="15"
      width="23px"
    />
    <Box
      height="23px"
      key="16"
      width="23px"
    />
    <Box
      height="23px"
      key="17"
      width="23px"
    />
    <Box
      height="23px"
      key="18"
      width="23px"
    />
    <Box
      height="23px"
      key="19"
      width="23px"
    />
    <Box
      height="23px"
      key="20"
      width="23px"
    />
  </Box>
  <Box
    display="flex"
    key="6"
  >
    <Box
      height="23px"
      key="0"
      width="23px"
    />
    <Box
      height="23px"
      key="1"
      width="23px"
    />
    <Box
      height="23px"
      key="2"
      width="23px"
    />
    <Box
      height="23px"
      key="3"
      width="23px"
    />
    <Box
      height="23px"
      key="4"
      width="23px"
    />
    <Box
      height="23px"
      key="5"
      width="23px"
    />
    <Box
      height="23px"
      key="6"
      width="23px"
    />
    <Box
      height="23px"
      key="7"
      width="23px"
    />
    <Box
      height="23px"
      key="8"
      width="23px"
    />
    <Box
      height="23px"
      key="9"
      width="23px"
    />
    <Box
      height="23px"
      key="10"
      width="23px"
    />
    <Box
      height="23px"
      key="11"
      width="23px"
    />
    <Box
      height="23px"
      key="12"
      width="23px"
    />
    <Box
      height="23px"
      key="13"
      width="23px"
    />
    <Box
      height="23px"
      key="14"
      width="23px"
    />
    <Box
      height="23px"
      key="15"
      width="23px"
    />
    <Box
      height="23px"
      key="16"
      width="23px"
    />
    <Box
      height="23px"
      key="17"
      width="23px"
    />
    <Box
      height="23px"
      key="18"
      width="23px"
    />
    <Box
      height="23px"
      key="19"
      width="23px"
    />
    <Box
      height="23px"
      key="20"
      width="23px"
    />
  </Box>
  <Box
    display="flex"
    key="7"
  >
    <Box
      height="23px"
      key="0"
      width="23px"
    />
    <Box
      height="23px"
      key="1"
      width="23px"
    />
    <Box
      height="23px"
      key="2"
      width="23px"
    />
    <Box
      height="23px"
      key="3"
      width="23px"
    />
    <Box
      height="23px"
      key="4"
      width="23px"
    />
    <Box
      height="23px"
      key="5"
      width="23px"
    />
    <Box
      height="23px"
      key="6"
      width="23px"
    />
    <Box
      height="23px"
      key="7"
      width="23px"
    />
    <Box
      height="23px"
      key="8"
      width="23px"
    />
    <Box
      height="23px"
      key="9"
      width="23px"
    />
    <Box
      height="23px"
      key="10"
      width="23px"
    />
    <Box
      height="23px"
      key="11"
      width="23px"
    />
    <Box
      height="23px"
      key="12"
      width="23px"
    />
    <Box
      height="23px"
      key="13"
      width="23px"
    />
    <Box
      height="23px"
      key="14"
      width="23px"
    />
    <Box
      height="23px"
      key="15"
      width="23px"
    />
    <Box
      height="23px"
      key="16"
      width="23px"
    />
    <Box
      height="23px"
      key="17"
      width="23px"
    />
    <Box
      height="23px"
      key="18"
      width="23px"
    />
    <Box
      height="23px"
      key="19"
      width="23px"
    />
    <Box
      height="23px"
      key="20"
      width="23px"
    />
  </Box>
  <Box
    display="flex"
    key="8"
  >
    <Box
      height="23px"
      key="0"
      width="23px"
    />
    <Box
      height="23px"
      key="1"
      width="23px"
    />
    <Box
      height="23px"
      key="2"
      width="23px"
    />
    <Box
      height="23px"
      key="3"
      width="23px"
    />
    <Box
      height="23px"
      key="4"
      width="23px"
    />
    <Box
      height="23px"
      key="5"
      width="23px"
    />
    <Box
      height="23px"
      key="6"
      width="23px"
    />
    <Box
      height="23px"
      key="7"
      width="23px"
    />
    <Box
      height="23px"
      key="8"
      width="23px"
    />
    <Box
      height="23px"
      key="9"
      width="23px"
    />
    <Box
      height="23px"
      key="10"
      width="23px"
    />
    <Box
      height="23px"
      key="11"
      width="23px"
    />
    <Box
      height="23px"
      key="12"
      width="23px"
    />
    <Box
      height="23px"
      key="13"
      width="23px"
    />
    <Box
      height="23px"
      key="14"
      width="23px"
    />
    <Box
      height="23px"
      key="15"
      width="23px"
    />
    <Box
      height="23px"
      key="16"
      width="23px"
    />
    <Box
      height="23px"
      key="17"
      width="23px"
    />
    <Box
      height="23px"
      key="18"
      width="23px"
    />
    <Box
      height="23px"
      key="19"
      width="23px"
    />
    <Box
      height="23px"
      key="20"
      width="23px"
    />
  </Box>
  <Box
    display="flex"
    key="9"
  >
    <Box
      height="23px"
      key="0"
      width="23px"
    />
    <Box
      height="23px"
      key="1"
      width="23px"
    />
    <Box
      height="23px"
      key="2"
      width="23px"
    />
    <Box
      height="23px"
      key="3"
      width="23px"
    />
    <Box
      height="23px"
      key="4"
      width="23px"
    />
    <Box
      height="23px"
      key="5"
      width="23px"
    />
    <Box
      height="23px"
      key="6"
      width="23px"
    />
    <Box
      height="23px"
      key="7"
      width="23px"
    />
    <Box
      height="23px"
      key="8"
      width="23px"
    />
    <Box
      height="23px"
      key="9"
      width="23px"
    />
    <Box
      height="23px"
      key="10"
      width="23px"
    />
    <Box
      height="23px"
      key="11"
      width="23px"
    />
    <Box
      height="23px"
      key="12"
      width="23px"
    />
    <Box
      height="23px"
      key="13"
      width="23px"
    />
    <Box
      height="23px"
      key="14"
      width="23px"
    />
    <Box
      height="23px"
      key="15"
      width="23px"
    />
    <Box
      height="23px"
      key="16"
      width="23px"
    />
    <Box
      height="23px"
      key="17"
      width="23px"
    />
    <Box
      height="23px"
      key="18"
      width="23px"
    />
    <Box
      height="23px"
      key="19"
      width="23px"
    />
    <Box
      height="23px"
      key="20"
      width="23px"
    />
  </Box>
  <Box
    display="flex"
    key="10"
  >
    <Box
      height="23px"
      key="0"
      width="23px"
    />
    <Box
      height="23px"
      key="1"
      width="23px"
    />
    <Box
      height="23px"
      key="2"
      width="23px"
    />
    <Box
      height="23px"
      key="3"
      width="23px"
    />
    <Box
      height="23px"
      key="4"
      width="23px"
    />
    <Box
      height="23px"
      key="5"
      width="23px"
    />
    <Box
      height="23px"
      key="6"
      width="23px"
    />
    <Box
      height="23px"
      key="7"
      width="23px"
    />
    <Box
      height="23px"
      key="8"
      width="23px"
    />
    <Box
      height="23px"
      key="9"
      width="23px"
    />
    <Box
      height="23px"
      key="10"
      width="23px"
    />
    <Box
      height="23px"
      key="11"
      width="23px"
    />
    <Box
      height="23px"
      key="12"
      width="23px"
    />
    <Box
      height="23px"
      key="13"
      width="23px"
    />
    <Box
      height="23px"
      key="14"
      width="23px"
    />
    <Box
      height="23px"
      key="15"
      width="23px"
    />
    <Box
      height="23px"
      key="16"
      width="23px"
    />
    <Box
      height="23px"
      key="17"
      width="23px"
    />
    <Box
      height="23px"
      key="18"
      width="23px"
    />
    <Box
      height="23px"
      key="19"
      width="23px"
    />
    <Box
      height="23px"
      key="20"
      width="23px"
    />
  </Box>
  <Box
    display="flex"
    key="11"
  >
    <Box
      height="23px"
      key="0"
      width="23px"
    />
    <Box
      height="23px"
      key="1"
      width="23px"
    />
    <Box
      height="23px"
      key="2"
      width="23px"
    />
    <Box
      height="23px"
      key="3"
      width="23px"
    />
    <Box
      height="23px"
      key="4"
      width="23px"
    />
    <Box
      height="23px"
      key="5"
      width="23px"
    />
    <Box
      height="23px"
      key="6"
      width="23px"
    />
    <Box
      height="23px"
      key="7"
      width="23px"
    />
    <Box
      height="23px"
      key="8"
      width="23px"
    />
    <Box
      height="23px"
      key="9"
      width="23px"
    />
    <Box
      height="23px"
      key="10"
      width="23px"
    />
    <Box
      height="23px"
      key="11"
      width="23px"
    />
    <Box
      height="23px"
      key="12"
      width="23px"
    />
    <Box
      height="23px"
      key="13"
      width="23px"
    />
    <Box
      height="23px"
      key="14"
      width="23px"
    />
    <Box
      height="23px"
      key="15"
      width="23px"
    />
    <Box
      height="23px"
      key="16"
      width="23px"
    />
    <Box
      height="23px"
      key="17"
      width="23px"
    />
    <Box
      height="23px"
      key="18"
      width="23px"
    />
    <Box
      height="23px"
      key="19"
      width="23px"
    />
    <Box
      height="23px"
      key="20"
      width="23px"
    />
  </Box>
  <Box
    display="flex"
    key="12"
  >
    <Box
      height="23px"
      key="0"
      width="23px"
    />
    <Box
      height="23px"
      key="1"
      width="23px"
    />
    <Box
      height="23px"
      key="2"
      width="23px"
    />
    <Box
      height="23px"
      key="3"
      width="23px"
    />
    <Box
      height="23px"
      key="4"
      width="23px"
    />
    <Box
      height="23px"
      key="5"
      width="23px"
    />
    <Box
      height="23px"
      key="6"
      width="23px"
    />
    <Box
      height="23px"
      key="7"
      width="23px"
    />
    <Box
      height="23px"
      key="8"
      width="23px"
    />
    <Box
      height="23px"
      key="9"
      width="23px"
    />
    <Box
      height="23px"
      key="10"
      width="23px"
    />
    <Box
      height="23px"
      key="11"
      width="23px"
    />
    <Box
      height="23px"
      key="12"
      width="23px"
    />
    <Box
      height="23px"
      key="13"
      width="23px"
    />
    <Box
      height="23px"
      key="14"
      width="23px"
    />
    <Box
      height="23px"
      key="15"
      width="23px"
    />
    <Box
      height="23px"
      key="16"
      width="23px"
    />
    <Box
      height="23px"
      key="17"
      width="23px"
    />
    <Box
      height="23px"
      key="18"
      width="23px"
    />
    <Box
      height="23px"
      key="19"
      width="23px"
    />
    <Box
      height="23px"
      key="20"
      width="23px"
    />
  </Box>
  <Box
    display="flex"
    key="13"
  >
    <Box
      height="23px"
      key="0"
      width="23px"
    />
    <Box
      height="23px"
      key="1"
      width="23px"
    />
    <Box
      height="23px"
      key="2"
      width="23px"
    />
    <Box
      height="23px"
      key="3"
      width="23px"
    />
    <Box
      height="23px"
      key="4"
      width="23px"
    />
    <Box
      height="23px"
      key="5"
      width="23px"
    />
    <Box
      height="23px"
      key="6"
      width="23px"
    />
    <Box
      height="23px"
      key="7"
      width="23px"
    />
    <Box
      height="23px"
      key="8"
      width="23px"
    />
    <Box
      height="23px"
      key="9"
      width="23px"
    />
    <Box
      height="23px"
      key="10"
      width="23px"
    />
    <Box
      height="23px"
      key="11"
      width="23px"
    />
    <Box
      height="23px"
      key="12"
      width="23px"
    />
    <Box
      height="23px"
      key="13"
      width="23px"
    />
    <Box
      height="23px"
      key="14"
      width="23px"
    />
    <Box
      height="23px"
      key="15"
      width="23px"
    />
    <Box
      height="23px"
      key="16"
      width="23px"
    />
    <Box
      height="23px"
      key="17"
      width="23px"
    />
    <Box
      height="23px"
      key="18"
      width="23px"
    />
    <Box
      height="23px"
      key="19"
      width="23px"
    />
    <Box
      height="23px"
      key="20"
      width="23px"
    />
  </Box>
  <Box
    display="flex"
    key="14"
  >
    <Box
      height="23px"
      key="0"
      width="23px"
    />
    <Box
      height="23px"
      key="1"
      width="23px"
    />
    <Box
      height="23px"
      key="2"
      width="23px"
    />
    <Box
      height="23px"
      key="3"
      width="23px"
    />
    <Box
      height="23px"
      key="4"
      width="23px"
    />
    <Box
      height="23px"
      key="5"
      width="23px"
    />
    <Box
      height="23px"
      key="6"
      width="23px"
    />
    <Box
      height="23px"
      key="7"
      width="23px"
    />
    <Box
      height="23px"
      key="8"
      width="23px"
    />
    <Box
      height="23px"
      key="9"
      width="23px"
    />
    <Box
      height="23px"
      key="10"
      width="23px"
    />
    <Box
      height="23px"
      key="11"
      width="23px"
    />
    <Box
      height="23px"
      key="12"
      width="23px"
    />
    <Box
      height="23px"
      key="13"
      width="23px"
    />
    <Box
      height="23px"
      key="14"
      width="23px"
    />
    <Box
      height="23px"
      key="15"
      width="23px"
    />
    <Box
      height="23px"
      key="16"
      width="23px"
    />
    <Box
      height="23px"
      key="17"
      width="23px"
    />
    <Box
      height="23px"
      key="18"
      width="23px"
    />
    <Box
      height="23px"
      key="19"
      width="23px"
    />
    <Box
      height="23px"
      key="20"
      width="23px"
    />
  </Box>
  <Box
    display="flex"
    key="15"
  >
    <Box
      height="23px"
      key="0"
      width="23px"
    />
    <Box
      height="23px"
      key="1"
      width="23px"
    />
    <Box
      height="23px"
      key="2"
      width="23px"
    />
    <Box
      height="23px"
      key="3"
      width="23px"
    />
    <Box
      height="23px"
      key="4"
      width="23px"
    />
    <Box
      height="23px"
      key="5"
      width="23px"
    />
    <Box
      height="23px"
      key="6"
      width="23px"
    />
    <Box
      height="23px"
      key="7"
      width="23px"
    />
    <Box
      height="23px"
      key="8"
      width="23px"
    />
    <Box
      height="23px"
      key="9"
      width="23px"
    />
    <Box
      height="23px"
      key="10"
      width="23px"
    />
    <Box
      height="23px"
      key="11"
      width="23px"
    />
    <Box
      height="23px"
      key="12"
      width="23px"
    />
    <Box
      height="23px"
      key="13"
      width="23px"
    />
    <Box
      height="23px"
      key="14"
      width="23px"
    />
    <Box
      height="23px"
      key="15"
      width="23px"
    />
    <Box
      height="23px"
      key="16"
      width="23px"
    />
    <Box
      height="23px"
      key="17"
      width="23px"
    />
    <Box
      height="23px"
      key="18"
      width="23px"
    />
    <Box
      height="23px"
      key="19"
      width="23px"
    />
    <Box
      height="23px"
      key="20"
      width="23px"
    />
  </Box>
  <Box
    display="flex"
    key="16"
  >
    <Box
      height="23px"
      key="0"
      width="23px"
    />
    <Box
      height="23px"
      key="1"
      width="23px"
    />
    <Box
      height="23px"
      key="2"
      width="23px"
    />
    <Box
      height="23px"
      key="3"
      width="23px"
    />
    <Box
      height="23px"
      key="4"
      width="23px"
    />
    <Box
      height="23px"
      key="5"
      width="23px"
    />
    <Box
      height="23px"
      key="6"
      width="23px"
    />
    <Box
      height="23px"
      key="7"
      width="23px"
    />
    <Box
      height="23px"
      key="8"
      width="23px"
    />
    <Box
      height="23px"
      key="9"
      width="23px"
    />
    <Box
      height="23px"
      key="10"
      width="23px"
    />
    <Box
      height="23px"
      key="11"
      width="23px"
    />
    <Box
      height="23px"
      key="12"
      width="23px"
    />
    <Box
      height="23px"
      key="13"
      width="23px"
    />
    <Box
      height="23px"
      key="14"
      width="23px"
    />
    <Box
      height="23px"
      key="15"
      width="23px"
    />
    <Box
      height="23px"
      key="16"
      width="23px"
    />
    <Box
      height="23px"
      key="17"
      width="23px"
    />
    <Box
      height="23px"
      key="18"
      width="23px"
    />
    <Box
      height="23px"
      key="19"
      width="23px"
    />
    <Box
      height="23px"
      key="20"
      width="23px"
    />
  </Box>
  <Box
    display="flex"
    key="17"
  >
    <Box
      height="23px"
      key="0"
      width="23px"
    />
    <Box
      height="23px"
      key="1"
      width="23px"
    />
    <Box
      height="23px"
      key="2"
      width="23px"
    />
    <Box
      height="23px"
      key="3"
      width="23px"
    />
    <Box
      height="23px"
      key="4"
      width="23px"
    />
    <Box
      height="23px"
      key="5"
      width="23px"
    />
    <Box
      height="23px"
      key="6"
      width="23px"
    />
    <Box
      height="23px"
      key="7"
      width="23px"
    />
    <Box
      height="23px"
      key="8"
      width="23px"
    />
    <Box
      height="23px"
      key="9"
      width="23px"
    />
    <Box
      height="23px"
      key="10"
      width="23px"
    />
    <Box
      height="23px"
      key="11"
      width="23px"
    />
    <Box
      height="23px"
      key="12"
      width="23px"
    />
    <Box
      height="23px"
      key="13"
      width="23px"
    />
    <Box
      height="23px"
      key="14"
      width="23px"
    />
    <Box
      height="23px"
      key="15"
      width="23px"
    />
    <Box
      height="23px"
      key="16"
      width="23px"
    />
    <Box
      height="23px"
      key="17"
      width="23px"
    />
    <Box
      height="23px"
      key="18"
      width="23px"
    />
    <Box
      height="23px"
      key="19"
      width="23px"
    />
    <Box
      height="23px"
      key="20"
      width="23px"
    />
  </Box>
  <Box
    display="flex"
    key="18"
  >
    <Box
      height="23px"
      key="0"
      width="23px"
    />
    <Box
      height="23px"
      key="1"
      width="23px"
    />
    <Box
      height="23px"
      key="2"
      width="23px"
    />
    <Box
      height="23px"
      key="3"
      width="23px"
    />
    <Box
      height="23px"
      key="4"
      width="23px"
    />
    <Box
      height="23px"
      key="5"
      width="23px"
    />
    <Box
      height="23px"
      key="6"
      width="23px"
    />
    <Box
      height="23px"
      key="7"
      width="23px"
    />
    <Box
      height="23px"
      key="8"
      width="23px"
    />
    <Box
      height="23px"
      key="9"
      width="23px"
    />
    <Box
      height="23px"
      key="10"
      width="23px"
    />
    <Box
      height="23px"
      key="11"
      width="23px"
    />
    <Box
      height="23px"
      key="12"
      width="23px"
    />
    <Box
      height="23px"
      key="13"
      width="23px"
    />
    <Box
      height="23px"
      key="14"
      width="23px"
    />
    <Box
      height="23px"
      key="15"
      width="23px"
    />
    <Box
      height="23px"
      key="16"
      width="23px"
    />
    <Box
      height="23px"
      key="17"
      width="23px"
    />
    <Box
      height="23px"
      key="18"
      width="23px"
    />
    <Box
      height="23px"
      key="19"
      width="23px"
    />
    <Box
      height="23px"
      key="20"
      width="23px"
    />
  </Box>
  <Box
    display="flex"
    key="19"
  >
    <Box
      height="23px"
      key="0"
      width="23px"
    />
    <Box
      height="23px"
      key="1"
      width="23px"
    />
    <Box
      height="23px"
      key="2"
      width="23px"
    />
    <Box
      height="23px"
      key="3"
      width="23px"
    />
    <Box
      height="23px"
      key="4"
      width="23px"
    />
    <Box
      height="23px"
      key="5"
      width="23px"
    />
    <Box
      height="23px"
      key="6"
      width="23px"
    />
    <Box
      height="23px"
      key="7"
      width="23px"
    />
    <Box
      height="23px"
      key="8"
      width="23px"
    />
    <Box
      height="23px"
      key="9"
      width="23px"
    />
    <Box
      height="23px"
      key="10"
      width="23px"
    />
    <Box
      height="23px"
      key="11"
      width="23px"
    />
    <Box
      height="23px"
      key="12"
      width="23px"
    />
    <Box
      height="23px"
      key="13"
      width="23px"
    />
    <Box
      height="23px"
      key="14"
      width="23px"
    />
    <Box
      height="23px"
      key="15"
      width="23px"
    />
    <Box
      height="23px"
      key="16"
      width="23px"
    />
    <Box
      height="23px"
      key="17"
      width="23px"
    />
    <Box
      height="23px"
      key="18"
      width="23px"
    />
    <Box
      height="23px"
      key="19"
      width="23px"
    />
    <Box
      height="23px"
      key="20"
      width="23px"
    />
  </Box>
  <Box
    display="flex"
    key="20"
  >
    <Box
      height="23px"
      key="0"
      width="23px"
    />
    <Box
      height="23px"
      key="1"
      width="23px"
    />
    <Box
      height="23px"
      key="2"
      width="23px"
    />
    <Box
      height="23px"
      key="3"
      width="23px"
    />
    <Box
      height="23px"
      key="4"
      width="23px"
    />
    <Box
      height="23px"
      key="5"
      width="23px"
    />
    <Box
      height="23px"
      key="6"
      width="23px"
    />
    <Box
      height="23px"
      key="7"
      width="23px"
    />
    <Box
      height="23px"
      key="8"
      width="23px"
    />
    <Box
      height="23px"
      key="9"
      width="23px"
    />
    <Box
      height="23px"
      key="10"
      width="23px"
    />
    <Box
      height="23px"
      key="11"
      width="23px"
    />
    <Box
      height="23px"
      key="12"
      width="23px"
    />
    <Box
      height="23px"
      key="13"
      width="23px"
    />
    <Box
      height="23px"
      key="14"
      width="23px"
    />
    <Box
      height="23px"
      key="15"
      width="23px"
    />
    <Box
      height="23px"
      key="16"
      width="23px"
    />
    <Box
      height="23px"
      key="17"
      width="23px"
    />
    <Box
      height="23px"
      key="18"
      width="23px"
    />
    <Box
      height="23px"
      key="19"
      width="23px"
    />
    <Box
      height="23px"
      key="20"
      width="23px"
    />
  </Box>
</ForwardRef(Card2)>
`;
